torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -625,21 +626,26 @@ std::vector<ComputationClient::ComputationPtr> PjRtComputationClient::Compile(
           device_assignment);
     }
 
-    std::unique_ptr<xla::PjRtLoadedExecutable> executable;
+    absl::StatusOr<std::unique_ptr<xla::PjRtLoadedExecutable>> maybe_executable;
     if (runtime::sys_util::GetEnvBool("XLA_STABLEHLO_COMPILE", false)) {
       // Convert HLO to StableHLO for PjRt client compilation.
       mlir::MLIRContext context;
       mlir::ModuleOp mlir_module =
           mlir::ModuleOp::create(mlir::UnknownLoc::get(&context));
       ConvertHloToStableHlo(instance.computation.mutable_proto(), &mlir_module);
-      executable =
-          client_->CompileAndLoad(mlir_module, compile_options).value();
+      maybe_executable = client_->CompileAndLoad(mlir_module, compile_options);
       StableHloCompileCounter()->AddValue(1);
     } else {
-      executable =
-          client_->CompileAndLoad(instance.computation, compile_options)
-              .value();
+      maybe_executable =
+          client_->CompileAndLoad(instance.computation, compile_options);
     }
+    if (!maybe_executable.ok()) {
+      // This will automatically raise a Python ValueError exception.
+      // See https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html.
+      throw std::invalid_argument(
+          std::string(maybe_executable.status().message()));
+    }
+    auto executable = std::move(maybe_executable).value();
```

Comment:
nit: `.value()` has a check for validity built in which is not need here.

Maybe just access using auto executable = std::move(*maybe_executable)

Reply:
Sorry I forgot to put this in draft mode. I plan to add tests and then ask for review.

There's one benefit of `std::move(maybe_executable).value()` - it enables warning on use of `maybe_executable` later. `std::move(*maybe_executable)` doesn't do that. I can switch to `*std::move(maybe_execuable)` though.

Reply:
Sounds good. Thanks!